components/kyma-environment-broker/internal/process/update/btp_operator_overrides.go

```diff
@@ -70,7 +72,8 @@ func (s *BTPOperatorOverridesStep) Run(operation internal.UpdatingOperation, log
 
 func (s *BTPOperatorOverridesStep) setBTPOperatorOverrides(c *reconcilerApi.Component, operation internal.UpdatingOperation) error {
```

Comment:
check if clusterId is not empty,
If it is empty, use kubeconfig to fetch it, then
  operation.clusterId = fetchedClusterID
  operationManager.save(operation)